### PR TITLE
fix various problem with compilation error reporting

### LINKED ISSUE #868: Bug: Viewer mode fails to show correct grammar text when graph error occurs
### Similar issues

- [X] I have searched and found no existing similar issues

### How are you using Studio?

Legend Studio

### Current and expected behavior

During a graph building error, viewer mode stays in form mode and reports graph building error. When user tries to investigate and goes to text mode, the grammar is shown from the state of the pure model at the time of the error. Therefore viewer mode is showing an incorrect grammar. It should show the grammar using the original entities when loading. 

### Steps to reproduce

Have a project with the following grammar (consider using model loader). Create a review and merge this. Then try to view the project in viewer mode, and toggle `text-mode`. The grammar is wrong.

### Model data

```
Class model::NewClass extends model::NewClass2
{
  prop1: String[1];
}


###Mapping
Mapping model::NewMapping
(
  model::NewClass: Pure
  {
  }
)
```

### Environment

```text
Chrome
```

### Possible solution and workaround

_No response_

### Contribution

- [ ] I would like to work on the fix for this issue

### PR DESCRIPTION
**Pull request by akphi**
## Summary

- [x] Closes #868 
- [x] Fix a bug where on graph compilation failure, we got change detection status shown as `change detection halted` which would never get updated (see screenshot below)

https://user-images.githubusercontent.com/13574879/172237095-37ca9f42-9d4d-468b-b2e3-fca86934a396.mov

## How did you test this change?

- [ ] Test(s) added
- [x] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

#### Change detection halted bug

https://user-images.githubusercontent.com/13574879/172237193-745222ab-e38c-484a-b686-00aca79ebac3.mov

### Reveal compilation error in text-mode in viewer mode - #868 

https://user-images.githubusercontent.com/13574879/172237184-c7dc13d7-969e-4c5b-b230-d9b7d1f7f6b0.mov

### CODE REVIEW

**Comment by MauricioUyaguari** [comment]
@akphi screenshot of project creation modal ?

**Comment by akphi** [comment]
@MauricioUyaguari just added some padding and change the font so it looks a little more highlighted

Screen Shot 2022-06-06 at 4 27 06 PM

File: packages/legend-studio/src/stores/ViewerStore.ts [hunk 1]
```diff
@@ -450,47 +375,131 @@ export class ViewerStore {
...
+            (yield this.editorStore.graphManagerState.graphManager.entitiesToPureCode(
+              entities,
+            )) as string;
+          yield flowResult(
+            this.editorStore.grammarTextEditorState.setGraphGrammarText(
+              editorGrammar,
+            ),
+          );
+        } catch (error2) {
+          assertErrorThrown(error2);
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/stores/ViewerStore.ts:485]: should we continue if it fails to convert to pure text. 
Almost falls into a similar category of `if something goes wrong with de-serialization, we can't really do anything but to alert`.

**Review by MauricioUyaguari — APPROVED** [review]

File: packages/legend-studio/src/stores/ViewerStore.ts (hunk 1 above)
Comment (akphi) [reply packages/legend-studio/src/stores/ViewerStore.ts]: Nice catch! Updated

**Pull request merged**